core/src/main/java/io/questdb/cairo/TxnScoreboardV2.java

```diff
@@ -0,0 +1,262 @@
+        updateMax(txn);
+
+        if (getActiveReaderCount() == 0) {
+            return false;
+        }
+
+        long scanCount = readScanCount();
+        for (int i = 0; i < scanCount; i++) {
+            long lockedTxn = Unsafe.getUnsafe().getLongVolatile(null, entriesMem + (long) i * Long.BYTES);
+            if (lockedTxn > UNLOCKED && lockedTxn < txn) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean incrementTxn(int id, long txn) {
+        long internalId = toInternalId(id);
+        assert internalId < entryScanCount;
+        if (Unsafe.getUnsafe().compareAndSwapLong(null, entriesMem + internalId * Long.BYTES, UNLOCKED, txn)) {
+            updateMaxReaderId(internalId);
+            incrementActiveReaderCount();
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean isMax(long txn) {
+        return txn >= getMax();
+    }
+
+    @Override
+    public boolean isRangeAvailable(long fromTxn, long toTxn) {
+        if (getActiveReaderCount() == 0) {
+            return true;
+        }
+
+        long scanCount = readScanCount();
+        for (int i = 0; i < scanCount; i++) {
+            long lockedTxn = Unsafe.getUnsafe().getLongVolatile(null, entriesMem + (long) i * Long.BYTES);
+            if (lockedTxn > UNLOCKED && lockedTxn >= fromTxn && lockedTxn < toTxn) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean isTxnAvailable(long txn) {
+        if (getActiveReaderCount() == 0) {
+            return true;
+        }
+
+        long scanCount = readScanCount();
+        for (int i = 0; i < scanCount; i++) {
+            long lockedTxn = Unsafe.getUnsafe().getLongVolatile(null, entriesMem + (long) i * Long.BYTES);
+            if (lockedTxn == txn) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public long releaseTxn(int id, long txn) {
+        long internalId = toInternalId(id);
+        assert internalId < entryScanCount;
+        Unsafe.getUnsafe().putLongVolatile(null, entriesMem + internalId * Long.BYTES, UNLOCKED);
+        // It's too expensive to count how many readers are left for the txn, caller will have to do additional checks.
+        return 0;
+    }
+
+    public void setTableToken(TableToken tableToken) {
+        this.tableToken = tableToken;
+    }
+
+    private static int toInternalId(int id) {
+        return id + VIRTUAL_ID_COUNT;
+    }
+
+    private long getActiveReaderCount() {
+        return Unsafe.getUnsafe().getLongVolatile(null, activeReaderCountMem);
+    }
+
+    private long getMax() {
+        return Unsafe.getUnsafe().getLongVolatile(null, maxMem);
+    }
+
+    private void incrementActiveReaderCount() {
+        long val;
+        do {
+            val = Unsafe.getUnsafe().getLong(activeReaderCountMem);
```

Comment:
```suggestion
            val = Unsafe.getUnsafe().getLongVolatile(activeReaderCountMem);
```

Reply:
removed this method, inlined instead in https://github.com/questdb/questdb/pull/5444/commits/b9e552aac1e8796dad95dffb1f79062ccdb9b925